examples/basic.py

```diff
@@ -45,6 +30,26 @@ def on_event(event: EdgeEvent, event_data: MariNode | Frame):
 )
 def main(port: str | None, log_dir: str):
     """A basic example of using the MariLib library."""
+    logger_container = {"instance": None}
+
+    def on_event(event: EdgeEvent, event_data: MariNode | Frame):
+        logger = logger_container["instance"]
+        if not (logger and logger.active and event_data):
+            return
+
+        if event == EdgeEvent.NODE_JOINED:
+            # print(f"Node {event_data} joined")
+            # print("#", end="", flush=True)
+            logger.log_event(event_data.address, "NODE_JOINED")
```

Comment:
1. Why bring the on_event inside main?

2. I think it is better to call the logger from the library itself, instead of letting it to the application. On the other hand, it's true that all the other log calls are in the application. So maybe it should be handled in a separate issue. Could you create one to track this?